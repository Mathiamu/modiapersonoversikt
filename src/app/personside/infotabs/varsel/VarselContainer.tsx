import * as React from 'react';
import { isNotStarted } from '../../../../redux/restReducers/restResource';
import { BaseUrlsResponse } from '../../../../models/baseurls';
import { Varsel } from '../../../../models/varsel';
import { AppState } from '../../../../redux/reducers';
import { AsyncDispatch } from '../../../../redux/ThunkTypes';
import { hentBaseUrls } from '../../../../redux/restReducers/baseurls';
import { hentVarsel } from '../../../../redux/restReducers/varsel';
import { connect } from 'react-redux';
import PlukkRestData from '../ytelser/pleiepenger/PlukkRestData';
import VarselVisning from './VarselVisningKomponent';
import { RestResource } from '../../../../redux/restReducers/restResource';

interface StateProps {
    baseUrlResource: RestResource<BaseUrlsResponse>;
    varselResource: RestResource<Varsel[]>;
}

interface DispatchProps {
    hentBaseUrls: () => void;
    hentVarsel: (fødselsnummer: string) => void;
}

interface OwnProps {
    fødselsnummer: string;
}

type Props = StateProps & DispatchProps & OwnProps;

class VarselContainer extends React.PureComponent<Props> {
    componentDidMount() {
        if (isNotStarted(this.props.baseUrlResource)) {
            this.props.hentBaseUrls();
        }
        if (isNotStarted(this.props.varselResource)) {
            this.props.hentVarsel(this.props.fødselsnummer);
        }
    }

    render() {
        return (
            <PlukkRestData restResource={this.props.varselResource}>
                {data => <VarselVisning varsler={data} />}
            </PlukkRestData>
        );
    }
}

function mapStateToProps(state: AppState): StateProps {
    return {
        baseUrlResource: state.restResources.baseUrl,
        varselResource: state.restResources.brukersVarsler
    };
}

function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
    return {
        hentBaseUrls: () => dispatch(hentBaseUrls()),
        hentVarsel: (fødselsnummer: string) => dispatch(hentVarsel(fødselsnummer))
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(VarselContainer);
