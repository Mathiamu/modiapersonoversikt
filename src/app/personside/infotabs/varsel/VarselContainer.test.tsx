import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import VarselContainer from './VarslerContainer';

beforeEach(() => {
    Date.prototype.getTime = jest.fn(() => 0);
    Date.parse = jest.fn(() => 0);
});

test('Viser varselcontainer med alt innhold', () => {
    const container = renderer.create(
        <TestProvider>
            <VarselContainer />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
