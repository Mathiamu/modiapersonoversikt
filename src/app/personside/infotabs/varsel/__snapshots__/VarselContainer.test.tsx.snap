// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser varselcontainer med alt innhold 1`] = `
.c1 {
  font-weight: bold;
}

.c0 table {
  width: 100%;
  text-align: right;
}

.c0 table thead {
  text-transform: uppercase;
}

.c0 table thead th {
  margin-bottom: 0.5rem;
}

.c0 table thead th:nth-child(3) ~ th {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 table tbody tr {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1);
  margin-bottom: 0.5rem;
  display: block;
}

.c0 table tbody tr td:last-child {
  width: 100%;
}

.c0 table th,
.c0 table td {
  display: inline-block;
}

.c0 table th:not(:last-child),
.c0 table td:not(:last-child) {
  vertical-align: bottom;
  padding: 1rem 1rem 0;
}

.c0 table th:first-child,
.c0 table td:first-child {
  text-align: left;
  width: 20%;
}

.c0 table th:nth-child(2),
.c0 table td:nth-child(2) {
  width: 50%;
}

.c0 table th:nth-child(3),
.c0 table td:nth-child(3) {
  width: 30%;
}

.c2 li {
  display: inline-block;
}

.c2 li:not(:last-child):after {
  content: ',';
  margin-right: 0.5em;
}

<article>
  <h2
    className="visually-hidden"
    tabIndex={-1}
  >
    Varsler
  </h2>
  <div
    className="typo-normal"
  >
    <div
      className="c0"
    >
      <table
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              role="columnheader"
            >
              <p
                className="typo-element"
              >
                Dato
              </p>
            </th>
            <th
              role="columnheader"
            >
              <p
                className="typo-element"
              >
                Type
              </p>
            </th>
            <th
              role="columnheader"
            >
              <p
                className="typo-element"
              >
                Sendt i kanal
              </p>
            </th>
            <th
              role="columnheader"
            >
              <p
                className="typo-element"
              >
                Detaljer
              </p>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              <span
                className="c1"
              >
                P책minnelse om 책 sende meldekort
              </span>
            </td>
            <td
              role="cell"
            >
              <ul
                className="c2"
              >
                <li>
                  EPOST
                </li>
                <li>
                  SMS
                </li>
              </ul>
            </td>
            <td
              role="cell"
            >
              <div
                className="ekspanderbartPanel ekspanderbartPanel--lukket"
              >
                <button
                  aria-expanded={false}
                  className="ekspanderbartPanel__hode"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <div
                    className="ekspanderbartPanel__flex-wrapper"
                  >
                    <span
                      className="typo-element ekspanderbartPanel__heading"
                    >
                      Vis detaljer
                    </span>
                    <span
                      className="ekspanderbartPanel__indikator"
                    />
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              <span
                className="c1"
              >
                P책minnelse om 책 sende meldekort
              </span>
            </td>
            <td
              role="cell"
            >
              <ul
                className="c2"
              >
                <li>
                  EPOST
                </li>
              </ul>
            </td>
            <td
              role="cell"
            >
              <div
                className="ekspanderbartPanel ekspanderbartPanel--lukket"
              >
                <button
                  aria-expanded={false}
                  className="ekspanderbartPanel__hode"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <div
                    className="ekspanderbartPanel__flex-wrapper"
                  >
                    <span
                      className="typo-element ekspanderbartPanel__heading"
                    >
                      Vis detaljer
                    </span>
                    <span
                      className="ekspanderbartPanel__indikator"
                    />
                  </div>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</article>
`;
